### DISCUSSION TITLE
Azure Blob instead of AWS S3?

### DISCUSSION THREAD

**Discussion by fleejy** [outline/outline#1620] [Self Hosting] [answered]
Is it possible to configure it with Azure Blob storage rather than AWS S3?

**Comment by tommoor** [comment, marked as answer]
Yes, you can use Minio as a gateway to convert the S3 API's to use blob storage. Here's a guide:

https://cloudblogs.microsoft.com/opensource/2017/11/09/s3cmd-amazon-s3-compatible-apps-azure-storage/

**Reply by guillaumemeyer** [reply]
Looks like the Minio gateway is gone and is not a viable solution anymore:
- https://blog.min.io/deprecation-of-the-minio-gateway/
- https://github.com/minio/minio/issues/14331
Any plan to support Azure blob storage through another solution?

**Comment by guillaumemeyer** [comment]
Hope it will be useful for other Azure users...
As Minio is deprecating the gateway, I'm now using the [S3proxy](https://github.com/gaul/s3proxy) project.

**Reply by craftzneko** [reply]
@guillaumemeyer i have been having real difficulties getting it to work, dont suppose you have a few steps on how you setup s3proxy to work?

**Reply by guillaumemeyer** [reply]
I actually abandoned this option...
Using a self hosted S3 on Azure instead.
I also found an alternative (not tested) https://www.zenko.io/blog/port-s3-apps-to-azure/

**Reply by guillaumemeyer** [reply]
Now using Obsidian for my team...

**Reply by craftzneko** [reply]
Thanks, would love to see outline support Azure storage blobs but suspect it will never happen. Will take a look at zenko though. Cheers 👍

**Reply by guillaumemeyer** [reply]
Actually I would prefer to see Microsoft supporting the S3 protocol... Can't believe this is still not available.

**Reply by craftzneko** [reply]
Actually YES, you're 100% right, how is this not a thing yet. Speaking to a MS rep they tell us its because the backend was built before S3 became a standard and changing it now would affect millions of users but the reality is they don't want to do it as they could make a new storage type easily! Inevitably the people who suffer from this are the people like us, having to work around these decisions. Frustrating!

**Reply by guillaumemeyer** [reply]
Especially when Minio demonstrated that it was perfectly possible to build a thin compatibility layer...

**Comment by jmealo** [comment]
**For anyone finding this via search:** I've opened a more recent RFC for native Azure Blob support in #12148.  Would appreciate eyes there.
